### PR TITLE
fix: download button pointing to an incorrect binary on Windows arm64

### LINKED ISSUE #7239: Incorrect download suggested on Windows arm64 machines
### URL:

https://nodejs.org/en

### Browser Name:

Edge

### Browser Version:

131.0.2903.51

### Operating System:

Windows 11, 24H2, 26100.2314

### How to reproduce the issue:

1. Open nodejs.org on Windows arm64 machine (e.g. Surface Laptop, 7th edition)
2. Click Download
3. At the moment of writing, `node-v22.11.0-x64.msi` gets downloaded.

Interestingly, https://nodejs.org/en/download/prebuilt-installer correctly recognizes my machine ARM64 and this particular download button works as expected. This button, however, receives `bitness` as `arm64` (as opposed to just `64`), from `ReleaseContext`, which in turn appears to be fed with correct value by `BitnessDropdown` right here:

https://github.com/nodejs/nodejs.org/blob/c5345f551ea545cf9d04017204b17f3099940ada/apps/site/components/Downloads/Release/BitnessDropdown.tsx#L16-L29

### PR DESCRIPTION
**Pull request by wojtekmaj**
## Description

This PR changes the way `DownloadButton` calculates `bitness` which gets passed to `getNodeDownloadUrl`. 

Previously, this value was passed directly from `useDetectOS()`, leading to the issue mentioned in the original issue.

After this change, `bitness` is calculated similarly to how it's done in `BitnessDropdown`, so: by passing `architecture` and `bitness` to `getUserBitnessByArchitecture`:

https://github.com/nodejs/nodejs.org/blob/c5345f551ea545cf9d04017204b17f3099940ada/apps/site/components/Downloads/Release/BitnessDropdown.tsx#L16-L29

## Validation

See #7239 for reproduction steps.

I've validated the fix by navigating to [Vercel preview](https://nodejs-org-git-fork-wojtekmaj-patch-1-openjs.vercel.app/en) (linked by a bot below) and verifying the binary link (see bottom left corner):

image

## Related Issues

Closes #7239 

### Check List

- [ ] I have read the [Contributing Guidelines](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md) and made commit messages that follow the guideline.
- [ ] I have run `npm run format` to ensure the code follows the style guide.
- [ ] I have run `npm run test` to check if all tests are passing.
- [ ] I have run `npx turbo build` to check if the website builds without errors.
- [ ] I've covered new added functionality with unit tests if necessary.

### CODE REVIEW

**Review by avivkeller — APPROVED** [review]

**Comment by avivkeller** [comment]
> `Error: //#prettier: command (/home/runner/work/nodejs.org/nodejs.org/) /opt/hostedtoolcache/node/20.18.0/x64/bin/npm run prettier exited (1)`

Is this reproducible locally? If not, I'll restart the CI to see if it was a flake?

**Comment by AugustinMauroy** [comment]
> Is this reproducible locally? If not, I'll restart the CI to see if it was a flake?

No there are no issue with prettier. The code format is just not right.

@wojtekmaj could you do a `npm run format` to format the code and have a green CI

**Review by araujogui — APPROVED** [review]

**Comment by wojtekmaj** [comment]
Ah, sorry! I did that PR entirely on the go :D

**Comment by avivkeller** [comment]
No worries! Once all the checks pass, since this has approvals, I've added it to the merge queue

**Pull request merged**
